lib/basin-util.c

```diff
@@ -32,6 +34,21 @@ basin_find_resources (const gchar *content)
         }
     }
   xmlXPathFreeObject (objects);
+
+  /* from videos */
+  objects = xmlXPathEvalExpression ("//a[@data-libingester-asset-id]", context);
+  if (!xmlXPathNodeSetIsEmpty (objects->nodesetval))
+    {
+      xmlNodeSetPtr nodeset = objects->nodesetval;
+      for (int i = 0; i < nodeset->nodeNr; i++)
+        {
+          xmlChar *ekn_id = xmlGetProp (nodeset->nodeTab[i], "data-libingester-asset-id");
+          list = g_list_append (list, g_strdup (ekn_id));
```

Comment:
Also a pre-existing problem, but I didn't spot it last time: `g_list_append()` is inefficient when adding multiple items, use `g_list_prepend()` and `g_list_reverse()` afterwards if the order matters.

Reply:
Thanks!